pkg/planner/core/logical_plan_builder.go

```diff
@@ -7000,14 +7030,11 @@ func (b *PlanBuilder) buildRecursiveCTE(ctx context.Context, cte ast.ResultSetNo
 					// Build seed part plan.
 					saveSelect := x.SelectList.Selects
 					x.SelectList.Selects = x.SelectList.Selects[:i]
-					// We're rebuilding the seed part, so we pop the result we built previously.
```

Comment:
Can I ask why we need to remove this?

Reply:
The `popMap()` here is moved right after `b.buildSelect()` and `b.buildSetOpr()` in this function.
This should make it more clear and explicit that we directly drop the entries pushed by them.